### DISCUSSION TITLE
New hotkey to cycle through overlays (per category)

### DISCUSSION THREAD

**Discussion by Ouaz** [Keriew/augustus#550] [Ideas]
Augustus provides a really great QoL improvement: the "`Relative overlay`" hotkey, which allows to display the desired overlay, by pointing a building with the mouse cursor, then press the assigned hotkey. By the way, it should be added to `Julius` too, as it's really (really) useful, by removing a lot of tedious clicks in the sidebar.

But it does not work for all overlays (especially some new overlays implemented in Augustus).

So, could be it possible to add a new hotkey that allows to cycle through overlays contained in a category? Let's say the [O] key.

Here's how it would work:

![cycle_overlay](https://user-images.githubusercontent.com/11946570/146093061-3c3b3d4a-da64-4a1b-be99-30cbb68403f9.gif)

Another example: the player presses [K] on a Tavern to display the "`Tavern`" overlay, and then [O] to cycle through overlays contained in the "`Entertainment`" overlay category (`"Overall", "Tavern", "Theater", "Amphitheater", "Arena", "Colosseum", "Hippodrome"`, and so on). The overlay button title changes according the displayed overlay when using [O], so the players always know which overlay it is.

No need anymore to click each time on the overlay button in the sidebar, or to search for a corresponding building on the map to use K, when the player wants to do a full check-up of the city (Risks, Education, Health, etc).

Thanks for reading!

**Comment by Riww** [comment]
Great idea! I often click through all the overlays, this would save so much time!

**Comment by kyklish** [comment]
I contributed `Relative Overlay` to `Augustus`.

> it should be added to `Julius` too

*bvschaik* (author of `Julius`) don't want this, I wrote him. `Julius` is pure remake of original `Caesar 3` with no new features.

> But it does not work for all overlays

@Ouaz write example please. Some building belongs to several overlays. Maybe some buildings don't belongs to any overlay.

I can wrote `Cycle Overlay`, but I need approval from @Keriew (will you accept such PR)?

**Reply by Keriew** [reply]
Sure, if you code it, I'll gladly add it.

**Comment by Ouaz** [comment]
> bvschaik (author of Julius) don't want this, I wrote him. Julius is pure remake of original Caesar 3 with no new features.

Oh, well, OK. I consider the "relative overlay" hotkey like a QoL improvement, not a new feature modifying Caesar 3 vanilla.^^
Julius has already the "clone building under cursor" hotkey which is the same kind of QoL improvement.

> write example please. Some building belongs to several overlays. Maybe some buildings don't belongs to any overlay.

This suggestion was made months ago, some relative overlays have been added since then (warehouses for example).

But for example, "Global" education and "Global" entertainment overlays can't be accessed neither by an overlay shortcut or a relative overlay. Same for "Sentiment" overlay.
So rather than adding a hotkey for any possible overlay (too many options kills the purpose of options), I thought it would be interesting to be able to cycle through overlays in the same category, via an unique hotkey.

**Comment by kyklish** [comment]
Deer @crudelios, we need your opinion about new hotkeys.
